test/libsolidity/syntaxTests/using/library_functions.sol

```diff
@@ -0,0 +1,20 @@
+library L {
+    function publicFunction(uint) public pure {}
+    function externalFunction(uint) external pure {}
+    function internalFunction(uint) internal pure {}
+}
+
+contract C {
+    using {L.publicFunction} for uint;
+    using {L.externalFunction} for uint;
+    using {L.internalFunction} for uint;
```

Comment:
Is there a semantic difference between this and 
```
using {L.publicFunction, L.externalFunction, L.internalFunction} for uint;
```
It might be useful to add tests covering this behaviour with other `using for` styles.

Reply:
There is no semantic difference. I guess I did it this way because I was testing individual behavior and it was easier to comment one or another. I think more tests won't hurt, right? I am going to add a version with the list.

Reply:
Hmmm, I'm sure the list is covered elsewhere though, so one might complain about duplicate tests :man_shrugging: 
Anyway, I'm not sure, but I shouldn't these tests be a part of the `semanticTests` suite? I thought `syntaxTests` were more for syntax errors... @ekpyron @cameel?

Reply:
I could not find a duplicate of that case yet...
I assumed this would be `syntaxTests` because the problem is about the allowed/disallowed types in the solidity expression rather than correct/incorrect result being produced. Does that make sense?

Reply:
You're both kind of right :-). This being disallowed is a syntactic issue - so having syntax tests for this makes sense. But also since this *wasn't* allowed before, we for sure don't have any tests for actually checking that this produces valid results, i.e. no semantics tests about calling such functions as library functions. So we should also have semantics tests about this.